resources/recursor_install_rhel.rb

```diff
@@ -60,27 +52,20 @@
     action :create
     not_if { new_resource.debug }
   end
-end
 
-action :uninstall do
-  yum_package 'pdns-recursor' do
-    action :remove
+  if new_resource.upgrade
+    yum_package 'pdns-recursor' do
+      action :upgrade
```

Comment:
I don't understand why you can set the action `:upgrade` with the version?
something like that:
```
yum_package 'pdns-recursor' do
    version new_resource.version unless new_resource.version.nil?
    action new_resource.allow_upgrade ? :upgrade : :install
end
```
No matter what (cf: yum_package definition)
```
:upgrade
    Install a package and/or ensure that a package is the latest version. This action will ignore the version attribute.
```
Again not the maintainers of the cookbook, just my point of view.